### PR TITLE
Update DG for Btaykb

### PR DESCRIPTION
**Pull request by Btaykb**
Update DG to add in Comment feature implementation details.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -194,6 +194,39 @@ Each field implements a `Comparable` interface.
...
+#### Aspect: How to remove a comment 
+####1. Alternative 1 (Current choice)
+Similar to how `Status` was implemented, upon user input of an empty comment (i.e `comment 1`), the comment of the 
+person at the particular index, in this case the first person, will be removed 
+- Pros: Can contain all logic related to adding and removing a `comment` all in the `CommentCommand` and 
+  `CommentCommandParser` classes. 
+- Cons: Not consistent with how the other fields like `Module` and `Person` are removed. These fields are removed 
+  via an additional command that specifically handles their deletion (i.e. `deletemodule` and `delete`).
+
+####2. Alternative 2 (Current)
```
Comment (fantablack) [inline docs/DeveloperGuide.md:223]: `(Current)` should be removed as Alternative 2 is not the current implementation

**Review by fantablack — COMMENTED** [review]
Implementation write up is seems alright, and I like the Alternatives section. However, there seems to be a duplicate `(Current)` for Alternative 2. 

As for the sequence diagram, I don't see any glaring issues right off the bat but it would be awesome if other reviewers could take a look at it too in case I missed anything out.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Btaykb) [reply docs/DeveloperGuide.md:223]: Oh yes, missed that. Thanks!

**Review by bakano98 — COMMENTED** [review]
LGTM. But would it be better if you changed `d:CommentCommand` to `c:CommentCommand`? More intuitive

**Pull request merged**